### PR TITLE
Update PROJECT in accordance with operator-sdk 1.23.0

### PR DESCRIPTION
**Pull request by Jaanki**
to use kubebuilder v3. This is in line with the config new operator-sdk
framework autogenerates as evident from

https://github.com/Jaanki/submariner-operator/pull/10/files#diff-d1fb402f1269ba7396dca9cbd03b44ac055874669bce1d0f239fa2f54ea3584b

This also requires updating operator-sdk and kustomize versions. Other
changes made are:

1. Install KUSTOMIZE only if it is not locally present using the install
   script provided at its repo. Remove our own provided script.
2. Install controller tools only if it is not locally present.
3. Add make targets to install kustomize, operator-sdk and controller
   tools. This will allow users to install these versions easily.
4. Don't create packagemanifests as bundle is the recommended way to
   ship the operator.

Signed-off-by: Janki Chhatbar <jchhatba@redhat.com>

### CODE REVIEW

**Review by skitt — DISMISSED** [review]

**Comment by Jaanki** [comment]
Need to update operator-sdk and kustomize versions.

File: Makefile [hunk 1]
```diff
@@ -17,10 +17,10 @@ endif
 
 ifneq (,$(DAPPER_HOST_ARCH))
 
-OPERATOR_SDK_VERSION := 1.0.1
+OPERATOR_SDK_VERSION := 1.23.0
 OPERATOR_SDK := $(CURDIR)/bin/operator-sdk
 
-KUSTOMIZE_VERSION := 3.10.0
+KUSTOMIZE_VERSION := 3.8.7
```
Comment (tpantelis) [inline Makefile]: Why was this version downgraded?

File: Makefile [hunk 2]
```diff
@@ -154,11 +154,17 @@ bin/%/submariner-operator: $(VENDOR_MODULES) main.go $(EMBEDDED_YAMLS)
...
 
-# Operator CRDs
+# Download controller-gen locally if not already downloaded.
+CONTROLLER_TOOLS_VERSION := 0.9.2
 $(CONTROLLER_GEN): $(VENDOR_MODULES)
 	mkdir -p $(@D)
-	$(GO) build -o $@ sigs.k8s.io/controller-tools/cmd/controller-gen
+	test -s $(CURDIR)/bin/controller-gen || $(GO) build -o $@ sigs.k8s.io/controller-tools/cmd/controller-gen
+	## TODO (Jaanki) Use go install instead
+	# $(GO) install sigs.k8s.io/controller-tools/cmd/controller-gen@v$(CONTROLLER_TOOLS_VERSION)
```
Comment (tpantelis) [inline Makefile:144]: Will this TODO be resolved in this PR? If not, what's preventing us from using `go install`?

File: Makefile [hunk 3]
```diff
@@ -233,25 +231,21 @@ golangci-lint: $(EMBEDDED_YAMLS)
...
-	sha256sum -c scripts/operator-sdk.sha256
+	mkdir -p bin && \
+	cd bin && \
+	curl -Lo $@ "https://github.com/operator-framework/operator-sdk/releases/download/v${OPERATOR_SDK_VERSION}/operator-sdk_linux_amd64" && \
+	curl -Lo checksums.txt.asc "https://github.com/operator-framework/operator-sdk/releases/download/v${OPERATOR_SDK_VERSION}/checksums.txt.asc" && \
+	curl -Lo checksums.txt "https://github.com/operator-framework/operator-sdk/releases/download/v${OPERATOR_SDK_VERSION}/checksums.txt" \
+	## TODO (Jaanki) Add checksums
+	#gpg --verify checksums.txt.asc checksums.txt
+	#gpgv --keyring scripts/operator-sdk-signing-keyring.gpg $@.asc $@
+	#sha256sum -c scripts/operator-sdk.sha256
```
Comment (tpantelis) [inline Makefile:224]: Will this TODO be resolved in this PR? If not, what's preventing it?

Comment (Jaanki) [reply Makefile:224]: I am not familiar with checksum tests using `checksum.txt` and `checksum.txt.asc`. @skitt is aware but he is on PTO. I suggest to not block the PR for this.

File: Makefile (hunk 2 above)
Comment (Jaanki) [reply Makefile:144]: Using `go install` fails with `go: [sigs.k8s.io/controller-tools/cmd/controller-gen@v0.9.2](http://sigs.k8s.io/controller-tools/cmd/controller-gen@v0.9.2): cannot query module due to -mod=vendor` error. I am not sure whats happening here. This is a non-blocker and should not block this PR.

**Review by tpantelis — DISMISSED** [review]

**Review by tpantelis — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -154,11 +136,17 @@ bin/%/submariner-operator: $(VENDOR_MODULES) main.go $(EMBEDDED_YAMLS)
 
 ci: $(EMBEDDED_YAMLS) golangci-lint markdownlint unit build images
 
-# Operator CRDs
+# Download controller-gen locally if not already downloaded.
+CONTROLLER_TOOLS_VERSION := 0.9.2
 $(CONTROLLER_GEN): $(VENDOR_MODULES)
 	mkdir -p $(@D)
-	$(GO) build -o $@ sigs.k8s.io/controller-tools/cmd/controller-gen
+	test -s $(CURDIR)/bin/controller-gen || $(GO) build -o $@ sigs.k8s.io/controller-tools/cmd/controller-gen
```
Comment (skitt) [inline Makefile]: If `controller-gen` should be built only if it doesn’t exist, the prerequisites should be updated; testing for the existence of the binary is what Make does.

```suggestion
$(CONTROLLER_GEN):
	mkdir -p $(@D)
	$(GO) build -o $@ sigs.k8s.io/controller-tools/cmd/controller-gen
```

File: Makefile [hunk 5]
```diff
@@ -182,11 +170,13 @@ is-semantic-version:
...
     endif
 
-# TODO: a workaround until this issue will be fixed https://github.com/kubernetes-sigs/kustomize/issues/4008
+## Download kustomize locally if not already downloaded.
+KUSTOMIZE_INSTALL_SCRIPT ?= "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"
 $(KUSTOMIZE):
 	mkdir -p $(@D)
-	#GOBIN=$(CURDIR)/bin GO111MODULE=on $(GO) get sigs.k8s.io/kustomize/kustomize/v3
-	scripts/kustomize/install_kustomize.sh $(KUSTOMIZE_VERSION) $(CURDIR)/bin
+	test -s $(CURDIR)/bin/kustomize || { curl -s $(KUSTOMIZE_INSTALL_SCRIPT) | bash -s -- $(subst v,,$(KUSTOMIZE_VERSION)) $(CURDIR)/bin; }
```
Comment (skitt) [inline Makefile]: Again, Make does this for us:

```suggestion
	{ curl -s $(KUSTOMIZE_INSTALL_SCRIPT) | bash -s -- $(subst v,,$(KUSTOMIZE_VERSION)) $(@D); }
```

File: Makefile [hunk 6]
```diff
@@ -233,25 +213,21 @@ golangci-lint: $(EMBEDDED_YAMLS)
...
-#     gpg --no-options -q --batch --no-default-keyring \
-#     --output scripts/operator-sdk-signing-keyring.gpg \
-#     --dearmor scripts/operator-sdk-signing-key.asc
 $(OPERATOR_SDK):
-	curl -Lo $@ "https://github.com/operator-framework/operator-sdk/releases/download/v${OPERATOR_SDK_VERSION}/operator-sdk-v${OPERATOR_SDK_VERSION}-x86_64-linux-gnu"
-	curl -Lo $@.asc "https://github.com/operator-framework/operator-sdk/releases/download/v${OPERATOR_SDK_VERSION}/operator-sdk-v${OPERATOR_SDK_VERSION}-x86_64-linux-gnu.asc"
-	gpgv --keyring scripts/operator-sdk-signing-keyring.gpg $@.asc $@
-	sha256sum -c scripts/operator-sdk.sha256
+	mkdir -p bin && \
+	cd bin && \
```
Comment (skitt) [inline Makefile]: Don’t hard-code the directory:

```suggestion
	mkdir -p $(@D) && \
	cd $(@D) && \
```

**Review by skitt — CHANGES_REQUESTED** [review]

**Review by skitt — APPROVED** [review]

**Pull request merged**